addons/stock/stock.py

```diff
@@ -2440,12 +2440,14 @@ def action_cancel(self, cr, uid, ids, context=None):
                     procurement_ids = procurement_obj.search(cr, uid, [('move_dest_id', '=', move.id)], context=context)
                     procurement_obj.cancel(cr, uid, procurement_ids, context=context)
             else:
-                if move.move_dest_id:
+                if move.move_dest_id and not move.origin_returned_move_id:
```

Comment:
So, this means that a move that was returned, should not propagate its cancels: why?